framework/src/main/java/org/checkerframework/common/wholeprograminference/WholeProgramInferenceScenesStorage.java

```diff
@@ -108,34 +107,34 @@ public void writeScenesToJaif() {
         }
         // Write scenes into .jaif files.
         for (String jaifPath : modifiedScenes) {
-            try {
-                AScene scene = scenes.get(jaifPath).clone();
-                removeIgnoredAnnosFromScene(scene);
-                new File(jaifPath).delete();
-                scene.prune();
-                if (!scene.isEmpty()) {
-                    // Only write non-empty scenes into .jaif files.
-                    IndexFileWriter.write(scene, new FileWriter(jaifPath));
-                }
-            } catch (IOException e) {
-                throw new UserError(
-                        "Problem while reading file in: "
-                                + jaifPath
-                                + ". Exception message: "
-                                + e.getMessage(),
-                        e);
-            } catch (DefException e) {
-                throw new BugInCF(e);
-            }
+            scenes.get(jaifPath).writeToJaif(jaifPath, annosToIgnore);
+        }
+        modifiedScenes.clear();
+    }
+
+    /**
+     * Writes the scenes out to .astub files. This method is an alternative to {@link
+     * #writeScenesToJaif}.
+     */
+    public void writeScenesToStub() {
```

Comment:
Reduce code duplication by making the `writeScenesToJaif` method take an `OutputFormat` argument.  (You can also rename the method if you want.)  Then you don't need `writeScenesToStub`.